src/System.Private.CoreLib/shared/System/IO/Stream.cs

```diff
@@ -189,6 +189,73 @@ private int GetCopyBufferSize()
             return bufferSize;
         }
 
+        public virtual void CopyTo(ReadOnlySpanAction<byte, object> callback, object state, int bufferSize)
+        {
+            if (callback == null) throw new ArgumentNullException(nameof(callback));
+
+            CopyTo(new WriteCallbackStream(callback, state), bufferSize);
+        }
+
+        public virtual Task CopyToAsync(Func<ReadOnlyMemory<byte>, object, CancellationToken, Task> callback, object state, int bufferSize, CancellationToken cancellationToken)
```

Comment:
Ditto

Reply:
We should revisit this signature. I think the func should return a ValueTask rather than a Task.

Reply:
Fixed => API changed
I have also changed the return type of `CopyToAsync` to `ValueTask`.